src/OSLikeStuff/task_scheduler.cpp

```diff
@@ -46,7 +45,7 @@ struct TaskManager {
 
 	std::array<Task, kMaxTasks> list{nullptr};
 	std::array<SortedTask, kMaxTasks> sortedList;
```

Comment:
it would probably be a good idea to document what state the various entries in these lists can be in, when, and why.